src/app/components/normal-component/count-down/count-down.component.spec.ts

```diff
@@ -21,19 +22,48 @@ describe('CountDownComponent', () => {
         it('should throw when setting chrono already started', () => {
             component.setDuration(1250);
             component.start();
-            const error: string = 'Assertion failure: Should not set a chrono that has already been started (undefined)! (extra data: undefined)';
-            expect(() => component.setDuration(1250)).toThrowError(error);
+            spyOn(Utils, 'assert').and.callFake((b: boolean, s: string) => {throw new Error('prout');});
+            const error: string = 'Should not set a chrono that has already been started (undefined)!';
+
+            expect(() => component.setDuration(1250)).toThrowError('prout');
+
+            expect(Utils.assert).toHaveBeenCalledOnceWith(false, error);
         });
     });
     describe('start', () => {
         it('should throw when starting without having been set', () => {
-            expect(() => component.start()).toThrowError('Assertion failure: Should not start a chrono that has not been set! (extra data: undefined)');
+            const error: string = 'Should not start a chrono that has not been set!';
+            let calledOnceWithError: boolean = false;
+            spyOn(Utils, 'assert').and.callFake((b: boolean, s: string) => {
+                if (b === false) {
+                    if (s === error) {
+                        calledOnceWithError = true;
+                    }
+                    throw new Error('prout');
+                }
+            });
+
+            expect(() => component.start()).toThrowError('prout');
+
+            expect(calledOnceWithError).toBeTrue();
```

Comment:
hmmmm j'suis pas convaincu de 13 lignes là où une devrait suffire. On n'a pas des outils pour tester logError qu'on peut adapter pour assert ? Histoire de faire ça en une seule ligne.

Reply:
Les test d'assert sont cassés, ils sont instable, des codes strictement identiques marchent pour l'un et pas pour l'autre, en ce fait, ils sont innaceptables. Ceci marche et est stable, en ce fait, c'est une amélioration et on devrait la prendre !

Reply:
Mais pourquoi tu fais pas ça de façon plus propre en t'inspirant du LogErrorMocker ? Un spyOn(assert).and.callFake(someMethod); puis expect(assert).toHaveBeenCalledOnceWith(...) ? Deux lignes vs 13, bien plus clean, bien plus court, pas de répétition, le même effort

Reply:
j'ai un entre deux là, mais j'sais pas à quel point il est plus stable !